jupyter_console/interactiveshell.py

```diff
@@ -597,3 +637,21 @@ def init_history(self):
         self.history_manager = ZMQHistoryManager(client=self.client)
         self.configurables.append(self.history_manager)
 
+    def get_source_and_reset(self):
```

Comment:
I had left this so that it could also fallback to the input-splitter. In fact, I added a flag "use_kernel_is_complete" to allow the frontend to use the built-in (old) version, and not ask the kernel. This could also be a workaround if a kernel hasn't implemented this yet, or if it is incorrect.